Tools/Page-Builder/page_builder/pagebuilder.py

```diff
@@ -0,0 +1,87 @@
+import markdown
+import json
+from jinja2 import Template
+
+
+class PageBuilder():
+    """
+    Receives a path to a JSON file containing metadata for a list of web pages
+    to be built. The metadata contains information on where the final HTML
+    file should be saved, which HTML template to use, and where the markdown
+    content for that page can be found.
+
+    Page metadata schema:
+    {
+        'Page Title': {
+            'template': either a path to the HTML template or null. If null,
+                        defaults to '../templates/page_template.html',
+            'content': path to a markdown file with the content to be rendered
+                       to HTML,
+            'pathout': path to where the final file should be saved
+        }
+    }
+    """
+    def __init__(self, pages):
+        """
+        This class is used to create generic web pages for PSL
+        Parameters
+        ----------
+        pages: dictionary containing information on all the pages to be created
+        Returns
+        -------
+        None
+        """
+        self.pages = pages
+        self.required_attributes = {'template', 'content', 'pathout'}
+
+    def build_pages(self):
+        """
+        Method to loop through all pages in JSON metadata and output rendered
+        HTML files
+        """
+        for page in self.pages:
+            attributes = set(self.pages[page].keys())
+            assert self.required_attributes.issubset(attributes)
+            title = page
+            pathout = self.pages[page]['pathout']
+            content = self.pages[page]['content']
+            template = self.pages[page]['template']
+            # use default template if template is null
+            if not template:
+                template = '../templates/page_template.html'
+
+            # read and convert markdown content to HTML
+            with open(content, 'r') as f:
+                md_text = f.read()
+            html_text = markdown.markdown(md_text)
+            # replce <code> tages with <i> tags
+            html_text = html_text.replace('<code>', '<i>')
+            html_text = html_text.replace('</code>', '</i>')
```

Comment:
@andersonfrailey why `<i>` instead of `<code>`?

Reply:
When you look at the website `<code>` shows up pink and I didn't think it looked good with the rest of the page. Thinking on it though I can just update the template to change the font color of elements wrapped in the `<code>` tag. I'll try that locally and see if that looks better.

Reply:
Ah, right. I had the same problem. Here's how I solved it: https://github.com/open-source-economics/PSL-Core/blob/master/Tools/Catalog-Builder/templates/model_template.html#L19-L22

I found the color choice in some blog post. I can dig it up if you are interested.

Reply:
Sweet thanks. I like this solution. I'll push a commit to make the change in a minute.